File: tests/judge/malicious-programs/fork_bomb.c
#include <unistd.h>
#include <stdio.h>

/* 恶意样例：fork bomb（设计 6.2 / 11 节第 5 条）。
 * 预期沙箱结果：cgroups v2 pids.max 抑制，进程树被 SIGKILL/资源耗尽终止，
 * 映射 RE/TLE；宿主机与其他沙箱不受影响。 */
int main(void) {
    while (1) {
        pid_t pid = fork();
        if (pid < 0) {
            /* pids.max 触顶：fork 失败即持续重试 */
            continue;
        }
        if (pid == 0) {
            pause();
        }
    }
    return 0;
}

File: tests/judge/malicious-programs/fork_bomb.py
# 恶意样例：fork bomb（Python 变体）。
# 预期沙箱结果：pids.max 抑制，RE/TLE；不逃逸宿主机。
import os
import time

while True:
    try:
        pid = os.fork()
    except OSError:
        continue
    if pid == 0:
        time.sleep(3600)
